src/app/services/coin.service.ts

```diff
@@ -0,0 +1,40 @@
+import { BehaviorSubject } from 'rxjs/BehaviorSubject';
+
+import { BaseCoin } from '../coins/basecoin';
+import { SkycoinCoin } from '../coins/skycoin.coin';
+import { TestCoin } from '../coins/test.coin';
+
+export class CoinService {
+
+  currentCoin: BehaviorSubject<BaseCoin> = new BehaviorSubject<BaseCoin>(null);
+
+  private coins: BaseCoin[] = [];
```

Comment:
It should be possible to consult the contents of this array. It will be used to fill in the list of available coins.

Reply:
@Senyoret1, what do you mean "to consult"? Do you mean to make it public?
This array is filled in in `loadAvailableCoins()` method.

Reply:
Yes, it have to be public for future use, sorry for the lack of clarity.

Reply:
Fixed.